libipl/p10/ipl0.C

```diff
@@ -620,6 +620,35 @@ static bool update_genesis_hwas_state(void)
 		}
 	}
 
+	// mark targets as non functional in the devtree
+        // if it is marked UNUSED in the MRW. This is a workaround for
+        // handling unused IOHS targets in Bonnell.	
+	pdbg_for_each_class_target("iohs", iohs_target)
```

Comment:
Update is required only for iohs targets under primary processor target.
Line 567 already iterating through processor target., you can embed this logic there .

Reply:
Since this a workarround , better to create the patch and apply for required box?

Reply:
> Update is required only for iohs targets under primary processor target.
Line 567 already iterating through processor target., you can embed this logic there .

+1 BMC is responsible for the primary processor only, the secondary processor will be taken care of by the hostboot and that should sync with the appreciated value to the BMC.

(Please check this, if it's not happening then hostboot needs to fix it else we will be in the infinite loop to boot)

>Since this a workarround , better to create the patch and apply for required box?

I think we cannot push a patch file in the openbmc build since it is a single image build and we don't have any way to say pick a patch file for a specific system that is we are using MACHINE as `p10bmc`

Reply:
> I think we cannot push a patch file in the openbmc build since it is a single image build and we don't have any way to say pick a patch file for a specific system that is we are using MACHINE as `p10bmc`

The proposed fix is also not system specific. can you clarify how you are going to get this proposed fix help to fix system specific?
.Since this is not actual fix , better to patch in the build now and once the actual fix is identified , fix can be merged.

Reply:
> > I think we cannot push a patch file in the openbmc build since it is a single image build and we don't have any way to say pick a patch file for a specific system that is we are using MACHINE as `p10bmc`
> 
> The proposed fix is also not system specific. can you clarify how you are going to get this proposed fix help to fix system specific? .Since this is not actual fix , better to patch in the build now and once the actual fix is identified , fix can be merged.

I thought this is a common solution for IOHS targets even though it is a workaround, is it not?

are you referring to ignoring unused targets in the device tree by either fixing them in the MRW xml or MRW parser? If so, how we can remove that target since it is unused but it is present, right?

I noticed a few IOHS targets are unused in the Rainer and Everest too, @devenrao do you validated these changes in all p10 systems? Just curious to get to know, why this is shown in the new system.

Reply:
looks still discussion is going on the permanent fix